### PR TITLE
added KRNN and Sandwich models and their example results based on Alpha360

### PR DESCRIPTION
**Pull request by yaxuan999**
## How Has This Been Tested?

- [ ] Pass the test by running: `pytest qlib/tests/test_all_pipeline.py` under upper directory of `qlib`.
- [x] If you are adding a new feature, test on your own test scripts.

## Screenshots of Test Results (if appropriate):
1. Pipeline test:
2. Your own tests:

## Types of changes

- [ ] Fix bugs
- [x] Add new feature
- [ ] Update documentation

### CODE REVIEW

**Comment by yaxuan999** [comment]
> 

@microsoft-github-policy-service agree [company="MSRA"]

**Comment by yaxuan999** [comment]
@microsoft-github-policy-service agree company="MSRA"

**Comment by yaxuan999** [comment]
Contributor License Agreement installed

**Comment by yaxuan999** [comment]
@microsoft-github-policy-service agree company="MSRA"

**Review by you-n-g — CHANGES_REQUESTED** [review]

File: qlib/contrib/model/pytorch_krnn.py [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+        # input shape: [batch_size, seq_len*input_dim]
+        # output shape: [batch_size, seq_len, input_dim]
+        x = x.view(x.shape[0], -1, self.input_dim).permute(0, 2, 1).to(self.device)
+        y = self.conv(x)  # [batch_size, output_dim, conved_seq_len]
+        y = y.permute(0, 2, 1)  # [batch_size, conved_seq_len, output_dim]
+
+        return y
+
+
+class KRNNEncoderBase(nn.Module):
```
[the remark this answers is not in the document]
Comment (yaxuan999) [reply qlib/contrib/model/pytorch_krnn.py:74]: changed

File: qlib/contrib/model/pytorch_krnn.py [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+        # [seq_len, batch_size, output_dim, num_dups]
+        hids = torch.stack(hids, dim=-1)
+        hids = hids.view(seq_len, batch_size, self.output_dim, self.dup_num)
+        hids = hids.mean(dim=3)
+        hids = hids.permute(1, 0, 2)
+
+        return hids
+
+
+class CNNKRNNEncoder(nn.Module):
```
[the remark this answers is not in the document]
Comment (yaxuan999) [reply qlib/contrib/model/pytorch_krnn.py:136]: changed

File: examples/benchmarks/Sandwich/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Sandwich
+* Code: [https://github.com/microsoft/FOST/blob/main/fostool/model/sandwich.py](https://github.com/microsoft/FOST/blob/main/fostool/model/sandwich.py)
```
[the remark this answers is not in the document]
Comment (yaxuan999) [reply examples/benchmarks/Sandwich/README.md:2]: the original code of this model is in FOST without any references, so I kept the link of FOST

**Pull request merged**